### PR TITLE
Check `base_of_exclude` attribute on Primitive

### PR DESCRIPTION
**Pull request by jeff-hernandez**
Currently, we check the `base_of_exclude` attribute on  Feature classes instead of Primitive classes which raises an error.

#### Example

```python
from woodwork.column_schema import ColumnSchema
import featuretools as ft
import woodwork as ww


class Abs(ft.primitives.TransformPrimitive):
    input_types = [ColumnSchema(semantic_tags={'numeric'})]
    name = 'abs'

    def get_function(self):
        return abs

class Sum(ft.primitives.AggregationPrimitive):
    input_types = [ColumnSchema(semantic_tags={'numeric'})]
    base_of_exclude = [Abs]
    name = 'sum'

    def get_function(self):
        return sum

es = ft.demo.load_mock_customer(return_entityset=True)

fd = ft.dfs(
    entityset=es,
    features_only=True,
    agg_primitives=[Sum],
    target_dataframe_name='customers',
)
```
```
AttributeError: 'AggregationFeature' object has no attribute 'base_of_exclude'
```

Related to #1739

### CODE REVIEW

File: featuretools/tests/primitive_tests/test_agg_feats.py [hunk 1]
```diff
@@ -200,6 +200,11 @@ def test_base_of_and_stack_on_heuristic(es, test_primitive):
     child.primitive.base_of = [test_primitive]
     assert check_stacking(test_primitive(), [child])
 
+    test_primitive.stack_on = []
+    child.primitive.base_of = []
+    child.primitive.base_of_exclude = [test_primitive]
+    assert not check_stacking(test_primitive(), [child])
```
Comment (rwedge) [inline featuretools/tests/primitive_tests/test_agg_feats.py:206]: if `stack_on` and `base_of` are both set to `[]`. `check_stacking` will return False even if `test_primitive` is not in `base_of_exclude`

Comment (jeff-hernandez) [reply featuretools/tests/primitive_tests/test_agg_feats.py:206]: Oh okay, interesting. I changed `stack_on` and `base_of` to None so they can return True. It kind of seems like whether `None` or `[]` they should return the same value.

Comment (rwedge) [reply featuretools/tests/primitive_tests/test_agg_feats.py]: I think `None` means no whitelist while `[]` means an empty whitelist, so nothing allowed

**Review by rwedge — APPROVED** [review]
LGTM

**Pull request merged**